File: BackendStage/src/main/java/com/example/backendstage/Services/StageService.java
package com.example.backendstage.Services;

import com.example.backendstage.Entity.Stage;

import com.example.backendstage.Repository.StageRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class StageService {

    @Autowired
    private StageRepository stageRepository;

    // Obtenir tous les stages
    public List<Stage> getAllStages() {
        return stageRepository.findAll();
    }

    // Obtenir un stage par ID
    public Optional<Stage> getStageById(String id) {
        return stageRepository.findById(id);
    }

    // Créer un nouveau stage
    public Stage createStage(Stage stage) {
        return stageRepository.save(stage);
    }

    // Mettre à jour un stage existant
    public Stage updateStage(String id, Stage stage) {
        if (stageRepository.existsById(id)) {
            stage.setId(id);
            return stageRepository.save(stage);
        } else {
            return null;
        }
    }

    // Supprimer un stage
    public void deleteStage(String id) {
        stageRepository.deleteById(id);
    }

    // Obtenir les stages par pipelineId
    public List<Stage> getStagesByPipelineId(String pipelineId) {
        return stageRepository.findByPipelineId(pipelineId);
    }
}

File: FrontEnd/manajero-front-admin/src/app/services/stage.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Stage } from '../models/stage.model';
import { Pipeline } from '../models/pipeline.model';

@Injectable({
  providedIn: 'root'
})
export class StageService {
    private baseUrl = 'http://localhost:8093/api/stages'; // Corriger le chemin de base

    constructor(private http: HttpClient) { }
  
    getAllStages(): Observable<Stage[]> {
      return this.http.get<Stage[]>(`${this.baseUrl}/getall`); // Ajouter le chemin pour obtenir tous les stages
    }
  
    getStageById(id: string): Observable<Stage> {
      return this.http.get<Stage>(`${this.baseUrl}/getstagebyid/${id}`); // Ajouter le chemin pour obtenir un stage par ID
    }
  
    createStage(stage: Stage): Observable<Stage> {
      return this.http.post<Stage>(`${this.baseUrl}/add`, stage); // Ajouter le chemin pour créer un stage
    }
  
    updateStage(id: string, stage: Stage): Observable<Stage> {
      return this.http.put<Stage>(`${this.baseUrl}/modify/${id}`, stage); // Ajouter le chemin pour mettre à jour un stage
    }
  
    deleteStage(id: string): Observable<void> {
      return this.http.delete<void>(`${this.baseUrl}/delete/${id}`); // Ajouter le chemin pour supprimer un stage
    }

    getStagesByPipelineId(pipelineId: string): Observable<Stage[]> {
      return this.http.get<Stage[]>(`${this.baseUrl}/GETstagesByPipelineId/${pipelineId}`); // Ajouter le chemin pour obtenir les stages par pipelineId
    }

}
